feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/networkManagement/add/networkDetails/AddNetworkViewModel.kt

```diff
@@ -0,0 +1,173 @@
+                progressConsumer = loadingState.progressConsumer(),
+                validationFailureTransformerCustom = { status, actions -> mapSaveCustomNetworkFailureToUI(resourceManager, status, actions) }
+            ) {
+                launch {
+                    executeSaving(validationPayload)
+                    loadingState.value = false
+                }
+            }
+        }
+    }
+
+    private suspend fun executeSaving(savingPayload: CustomNetworkPayload) {
+        val nodeName = resourceManager.getString(R.string.create_network_node_name, savingPayload.chainName)
+        val blockExplorerName = resourceManager.getString(R.string.create_network_block_explorer_name, savingPayload.chainName)
+        val blockExplorerNameAndUrl = savingPayload.blockExplorerUrl?.let { blockExplorerName to it }
+
+        val result = when (payload.mode) {
+            AddNetworkPayload.Mode.EVM -> interactor.createEvmNetwork(
+                chainId = savingPayload.evmChainId!!,
+                iconUrl = payload.prefilledData?.iconUrl,
+                nodeUrl = savingPayload.nodeUrl,
+                nodeName = nodeName,
+                chainName = savingPayload.chainName,
+                tokenSymbol = savingPayload.tokenSymbol,
+                blockExplorer = blockExplorerNameAndUrl,
+                coingeckoLink = savingPayload.coingeckoLinkUrl
+            )
+
+            AddNetworkPayload.Mode.SUBSTRATE -> interactor.createSubstrateNetwork(
+                iconUrl = payload.prefilledData?.iconUrl,
+                nodeUrl = savingPayload.nodeUrl,
+                nodeName = nodeName,
+                chainName = savingPayload.chainName,
+                tokenSymbol = savingPayload.tokenSymbol,
+                blockExplorer = blockExplorerNameAndUrl,
+                coingeckoLink = savingPayload.coingeckoLinkUrl,
+                coroutineScope = viewModelScope
+            )
+        }
+
+        result.onSuccess { router.finishCreateNetworkFlow() }
+            .onFailure {
+                Log.e(LOG_TAG, "Failed to save network", it)
+                showError(resourceManager.getString(R.string.common_something_went_wrong_title))
+            }
+    }
+
+    private fun getValidationSystem(): CustomNetworkValidationSystem {
+        return when (payload.mode) {
+            AddNetworkPayload.Mode.EVM -> interactor.getEvmValidationSystem(viewModelScope)
+            AddNetworkPayload.Mode.SUBSTRATE -> interactor.getSubstrateValidationSystem(viewModelScope)
+        }
+    }
+
+    private fun runAutofill() {
+        val autofillMixin = when (payload.mode) {
+            AddNetworkPayload.Mode.EVM -> autofillNetworkMetadataMixinFactory.evm(viewModelScope)
+            AddNetworkPayload.Mode.SUBSTRATE -> autofillNetworkMetadataMixinFactory.substrate(viewModelScope)
+        }
+
+        nodeUrlFlow
+            .drop(1)
+            .mapLatest { url -> autofillMixin.autofill(url) }
```

Comment:
I think it makes sense to perform some simple pre-validation of url to avoid attempts to init a conneciton on every url input change
We should also probably debounce it for some reasonable time like 500ms

Reply:
Okay, I'll add it